service/history/outbound_queue_factory.go

```diff
@@ -74,27 +74,26 @@ type outboundQueueFactory struct {
 
 func NewOutboundQueueFactory(params outboundQueueFactoryParams) QueueFactory {
 	rateLimiterPool := collection.NewOnceMap(
-		func(queues.StateMachineTaskTypeNamespaceIDAndDestination) quotas.RateLimiter {
+		func(queues.NamespaceIDAndDestination) quotas.RateLimiter {
 			// TODO: get this value from dynamic config.
 			return quotas.NewDefaultOutgoingRateLimiter(func() float64 { return 100.0 })
 		},
 	)
 	circuitBreakerPool := collection.NewOnceMap(
-		func(key queues.StateMachineTaskTypeNamespaceIDAndDestination) *gobreaker.TwoStepCircuitBreaker {
+		func(key queues.NamespaceIDAndDestination) *gobreaker.TwoStepCircuitBreaker {
 			// TODO: get circuit breaker settings from dynamic config.
 			return gobreaker.NewTwoStepCircuitBreaker(
 				gobreaker.Settings{
 					Name: fmt.Sprintf(
-						"circuit_breaker:%d.%s.%s",
-						key.StateMachineTaskType,
+						"circuit_breaker:%s:%s",
 						key.NamespaceID,
 						key.Destination,
 					),
 				},
 			)
 		},
 	)
-	grouper := queues.GrouperStateMachineNamespaceIDAndDestination{}
+	grouper := queues.GrouperNamespaceIDAndDestination{}
```

Comment:
I'm debating whether we also want to group tasks by this predicate.

Right now there are 3 task types on this queue:

1. callbacks
2. nexus start operation
3. nexus cancel operation

We want 2 and 3 to be grouped together and 1 to never be grouped with them.
Today coincidentally your change achieves want we want works because the destinations for callbacks and nexus operations are not going to overlap.

When we add a fourth task type, we'll want the able to control the grouping.

So I think we actually want to instead of removing the task type dimension, replace it with something that gives us extra control. That'll be a bit more difficult to set up because you'll want the grouping to be set where the task is defined, e.g. add it here https://github.com/temporalio/temporal/blob/794eccdb92210d427a15ab4840b1c7f2d11a2eae/service/history/hsm/tasks.go#L87 and https://github.com/temporalio/temporal/blob/794eccdb92210d427a15ab4840b1c7f2d11a2eae/service/history/tasks/state_machine_task.go#L64.

If you want to put in the effort for that, go ahead but otherwise leave the code that's in main.